### PR TITLE
Boilerpate config service

### LINKED ISSUE #17: Create makefile / docker-compose for the Configuration Service (etcd)
This will be done mirroring @GiancarloFusiello example pattern
- [ ] Create boilerplate service
- [ ] Add script to inject test configuration for storage

### PR DESCRIPTION
**Pull request by diversemix**
Closes #17

### CODE REVIEW

File: configuration/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    environment:
+      ETCDCTL_API: 3
+      ETCD_DEBUG: 1
+      ETCD_INITIAL_ADVERTISE_PEER_URLS: http://127.0.0.1:2380
+      ETCD_LISTEN_CLIENT_URLS: http://0.0.0.0:2379
+      ETCD_LISTEN_PEER_URLS: http://0.0.0.0:2380
+      ETCD_ADVERTISE_CLIENT_URLS: http://127.0.0.1:2379
+      ETCD_INITIAL_CLUSTER: default=http://127.0.0.1:2380
+    ports:
+      - 2379:2379
```
Comment (giorgiosironi) [inline configuration/docker-compose.yml]: always specify `ports` short syntax values as [strings](https://docs.docker.com/compose/compose-file/#ports)

File: configuration/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+version: "3.4"
+
+services:
+  etcd1:
```
Comment (giorgiosironi) [inline configuration/docker-compose.yml:4]: `1`, is there a `2`?

File: configuration/Makefile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+stop:
+	docker-compose down -v
+
+configuration:
+	docker-compose up etcd1
+
+health:
+	docker-compose exec etcd1 etcdctl endpoint health
```
Comment (giorgiosironi) [inline configuration/Makefile]: define as `healthcheck` in `docker-compose.yml` instead?

File: configuration/Makefile [hunk 4]
```diff
@@ -0,0 +1,8 @@
+stop:
+	docker-compose down -v
+
+configuration:
+	docker-compose up etcd1
```
Comment (giorgiosironi) [inline configuration/Makefile]: doesn't need the service name `etcd1` as it's the only one?

File: configuration/Makefile [hunk 5]
```diff
@@ -0,0 +1,8 @@
+stop:
+	docker-compose down -v
```
Comment (giorgiosironi) [inline configuration/Makefile]: there are no named volumes, so `-v` is unnecessary?
(data won't be removed by `make stop` as this doesn't touch the host folders mounted as volumes)

File: configuration/docker-compose.yml (hunk 2 above)
Comment (diversemix) [reply configuration/docker-compose.yml:4]: not yet ... but this seems a convention that people use. Happy to remove the indexing if you find it confusing?

File: configuration/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    image: gcr.io/etcd-development/etcd:v3.3
+    restart: always
```
Comment (NuclearRedeye) [inline configuration/docker-compose.yml]: I personally prefer 'unless-stopped' to 'always', but that's just a suggestion.

File: configuration/docker-compose.yml (hunk 2 above)
Comment (petereast) [reply configuration/docker-compose.yml:4]: I think `docker-compose` handles numbering of replicas itself: https://docs.docker.com/compose/compose-file/#replicas

File: configuration/docker-compose.yml (hunk 6 above)
Comment (giorgiosironi) [reply configuration/docker-compose.yml]: As long as they both deal with the reboot case

Comment (petereast) [reply configuration/docker-compose.yml]: nO JOeL iTs HIgH AvAILaBIlITy!!!!@!@!@#!@#

File: configuration/docker-compose.yml (hunk 2 above)
Comment (diversemix) [reply configuration/docker-compose.yml:4]: @petereast ... problem with that is that the env vars `ETCD..." need changing differently for each instance.

File: configuration/docker-compose.yml (hunk 6 above)
Comment (diversemix) [reply configuration/docker-compose.yml]: > The unless-stopped restart policy behaves the same as always with one exception. When a container is stopped and the server is rebooted or the Docker service is restarted, the container will not be restarted.

What do we think we should be using?

Comment (NuclearRedeye) [reply configuration/docker-compose.yml]: I'm always of the view that if an admin/user/somebody has taken the decision to stop a container, that it should remain stopped until explicitly started again.
Honestly though I don't think it matters that much and if we change our minds later it's a trivial change to make.

File: configuration/Makefile [hunk 7]
```diff
@@ -0,0 +1,15 @@
+stop:
+	docker-compose down
+
+configuration:
+	docker-compose up
+
+status:
+	docker-compose exec etcd1 etcdctl endpoint status
```
Comment (giorgiosironi) [inline configuration/Makefile:8]: now that it is an `healthcheck` this should just be `docker-compose ps`?

**Review by NuclearRedeye — APPROVED** [review]

File: configuration/docker-compose.yml (hunk 2 above)
Comment (giorgiosironi) [reply configuration/docker-compose.yml:4]: this docker-compose configuration is unsuitable for multiple node deployments (or at least we wouldn't use it that way)

**Review by will-byrne — APPROVED** [review]

**Review by giorgiosironi — APPROVED** [review]

File: configuration/Makefile (hunk 7 above)
Comment (diversemix) [reply configuration/Makefile:8]: The health check is different to the status though :thinking: ?

File: configuration/docker-compose.yml (hunk 2 above)
Comment (diversemix) [reply configuration/docker-compose.yml:4]: well its just a name :grin:

File: configuration/Makefile (hunk 7 above)
Comment (giorgiosironi) [reply configuration/Makefile:8]: Didn't notice that, then good call.

**Pull request merged**